### DISCUSSION TITLE
I want to feel safe displaying user-generated HTML

### DISCUSSION THREAD

**Discussion by aarongustafson** [WebWeWant/webwewant.fyi#58] [Wants]
Many web applications allow untrusted users to upload HTML. They often do this using a rich text editor. Then, they show this HTML to other users. Email applications like Gmail are like this. However, displaying user-generated HTML is challenging from a security point of view. Sometimes you can sanitize it on the server. Sometimes you can sanitize it on the client using something like [DOMPurify](https://www.npmjs.com/package/dompurify). However, no matter how you do it, it’s a hard problem.

It'd be great if browsers could do this for us since they know how to do it best.

<hr>

https://webwewant.fyi/wants/5ee582f557f49af84b6bb374/

**Comment by jjinux** [comment]
It'd also be really helpful from a performance point of view since I wouldn't need to distribute a large library to do this, and I'm sure it could be made faster via browser code.
